dashboard/app/controllers/application_controller.rb

```diff
@@ -356,6 +356,29 @@ def prevent_caching
     redirect_to lockout_path if Policies::ChildAccount::ComplianceState.locked_out?(current_user)
   end
 
+  # Check that the user has completed the LMS onboarding flow
+  protected def assert_lms_landing_policy
+    return unless current_user.authentication_options.first&.lti? && !current_user.lms_landing_opted_out
+
+    # URLs we should not redirect.
+    return if Set[
```

Comment:
Do we need to add the `user_session_path` to allow them to get to the link existing account page?

Reply:
No, because when they click to link an account, the user is signed out and therefore there is no current user.

Reply:
You're totally right, as long as we allow `finish_link` we're good!